database/src/service/read.rs

```diff
@@ -8,6 +8,8 @@ use std::{
 
 use crossbeam::channel::Receiver;
 
+use futures::channel::oneshot;
```

Comment:
`InfallibleOneshotReceiver` uses `futures`'s oneshot, not `tokio` - on purpose?

Reply:
no, is tokio's better?

Reply:
Not sure, `futures` is fine though